src/lib/image/squashfs/init.c

```diff
@@ -76,8 +93,41 @@ int _singularity_image_squashfs_init(struct image_object *image, int open_flags)
     /* if LAUNCH_STRING is present, figure out squashfs magic offset */
     p = strstr(buf, "hsqs");
     if ( p != NULL ) {
-        singularity_message(VERBOSE2, "File is a valid SquashFS image\n");
+        char *compression_type = NULL;
+
         image->offset = p - buf;
+
+        if ( ( image->offset + sizeof(struct squashfs_header) ) > sizeof(buf) ) {
+            singularity_message(VERBOSE, "File is not a valid SquashFS image\n");
+            return(-1);
+        }
+
+        singularity_message(VERBOSE2, "File is a valid SquashFS image\n");
+
+        s_header = (struct squashfs_header *)p;
+
+        switch( s_header->compression ) {
+        case SQUASHFS_LZMA:
+            compression_type = strdup("lzma");
+            break;
+        case SQUASHFS_LZO:
+            compression_type = strdup("lzo");
+            break;
+        case SQUASHFS_XZ:
+            compression_type = strdup("xz");
+            break;
+        case SQUASHFS_LZ4:
+            compression_type = strdup("lz4");
+            break;
+        default:
+            break;
+        }
+
+        if ( compression_type ) {
+            singularity_message(WARNING, "squashfs image was compressed with %s, if it failed to run, " \
+                                         "please contact image's author\n", compression_type);
```

Comment:
We shouldn't put things on two lines like this, I would say just make the text all one string.

Reply:
Ok, will apply changes